overlord/devicestate/devicemgr.go

```diff
@@ -1385,3 +1391,66 @@ func checkFDEFeatures(st *state.State, kernelInfo *snap.Info) error {
 	//       returns {"error":"hardware-unsupported"} we don't.
 	return nil
 }
+
+func (m *DeviceManager) runFDESetupHook(op string, params *boot.FdeSetupHookParams) ([]byte, error) {
```

Comment:
what's the state locking expecations of this, I fear it will be called with the lock taken usually? we need to be very careful about that

Reply:
it also doesn't affect install I think but if we have to use this to reseal we have the problem that this won't work  if the kernel snap is being manipulated (for example its security profiles changed) while we try to run the hook, we need to leave at least a TODO about that ? it's a variant of the bug where we can't run hook when the bases are being updated

Reply:
Indeed, handlers_install.go will call bootMakeBootable() with the lock held and that will in turn call runFDESetupHook. So this needs to be updated so that this is handled (and tested) correctly.

Reply:
I also added a TODO